src/main/java/io/kestra/plugin/influxdb/FluxQuery.java

```diff
@@ -0,0 +1,281 @@
+                    outputBuilder
+                        .rows(fetch.getLeft())
+                        .size(fetch.getRight());
+                    break;
+
+                case FETCH_ONE:
+                    Map<String, Object> row = this.fetchOne(tables);
+                    outputBuilder
+                        .row(row)
+                        .size(row != null ? 1 : 0);
+                    break;
+
+                case STORE:
+                    Pair<URI, Long> store = this.store(runContext, tables);
+                    outputBuilder
+                        .uri(store.getLeft())
+                        .size(store.getRight().intValue());
+                    break;
+
+                case NONE:
+                    outputBuilder.size(recordCount);
+                    break;
+            }
+
+            runContext.metric(Counter.of("records", recordCount));
+            return outputBuilder.total((long) tables.size()).build();
+        }
+    }
+
+    protected Pair<List<Map<String, Object>>, Integer> fetch(List<FluxTable> tables) {
+        List<Map<String, Object>> results = new ArrayList<>();
+
+        for (FluxTable table : tables) {
+            for (FluxRecord record : table.getRecords()) {
+                Map<String, Object> row = new HashMap<>();
+                record.getValues().forEach((key, value) -> {
+                    if (value != null) {
+                        row.put(key, value);
+                    }
+                });
+
+                results.add(row);
+            }
+        }
+
+        return Pair.of(results, results.size());
+    }
+
+    protected Map<String, Object> fetchOne(List<FluxTable> tables) {
+        if (tables.isEmpty() || tables.get(0).getRecords().isEmpty()) {
+            return null;
+        }
+
+        FluxRecord record = tables.get(0).getRecords().get(0);
+        Map<String, Object> row = new HashMap<>();
+
+        record.getValues().forEach((key, value) -> {
+            if (value != null) {
+                row.put(key, value);
+            }
+        });
+
+        return row;
+    }
+
+    protected Pair<URI, Long> store(RunContext runContext, List<FluxTable> tables) throws IOException {
+        File tempFile = runContext.workingDir().createTempFile(".ion").toFile();
+
+        try (var output = new BufferedWriter(new FileWriter(tempFile), FileSerde.BUFFER_SIZE)) {
+            Flux<Map<String, Object>> recordFlux = Flux.fromIterable(tables)
+                .flatMap(table -> Flux.fromIterable(table.getRecords()))
+                .map(record -> {
+                    Map<String, Object> row = new HashMap<>();
+                    record.getValues().forEach((key, value) -> {
+                        if (value != null) {
+                            row.put(key, value);
+                        }
+                    });
+                    return row;
+                });
+
+            Long count = FileSerde.writeAll(output, recordFlux).block();
```

Comment:
formatting seems off, maybe use a quick formatting in intellij ? (better than nothing)